traffic_ops/traffic_ops_golang/routes.go

```diff
@@ -34,12 +34,16 @@ var NoAuth = false
 
 // Routes returns the routes, and a catchall route for when no route matches.
 func Routes(d ServerData) ([]Route, http.Handler, error) {
+
 	routes := []Route{
 		//ASNs
 		{1.2, http.MethodGet, `asns-wip(\.json)?$`, ASNsHandler(d.DB), ASNsPrivLevel, Authenticated, nil},
 		//CDNs
 		{1.2, http.MethodGet, `cdns-wip(\.json)?$`, cdnsHandler(d.DB), CDNsPrivLevel, Authenticated, nil},
 		{1.2, http.MethodGet, `cdns/{name}/configs/monitoring(\.json)?$`, monitoringHandler(d.DB), MonitoringPrivLevel, Authenticated, nil},
+		// Delivery services
+		{1.3, http.MethodGet, "deliveryservices/{xml-id}/urisignkeys$", urisignkeysHandler(d.DB, d.Config), PrivLevelAdmin, Authenticated, nil},
+		{1.3, http.MethodPost, "deliveryservices/{xml-id}/urisignkeys$", assignDeliveryServiceUriKeysKeysHandler(d.DB, d.Config), PrivLevelAdmin, Authenticated, nil},
```

Comment:
question:

looking at these 2 routes, it looks like I can "get" urisignkeys for a delivery service AND "create" urisignkeys for a delivery service.....

but can I update existing urisignkeys? or does it not work that way?

Reply:
actually, never mind, it looks like you 'get ds urisignkeys' with the GET and 'assign ds urisignkeys' with the POST.

do you think you can rename those hanlders to getDeliveryServiceUriSignKeys and assignDeliveryServiceUriSignKeysHandler so it's clear what each handler does? (you have a typo in that 2nd handler)

Reply:
I can add an update later but in reality if you use the POST to the same delivery service, the new data will overwrite the keys that are there now.

Reply:
make sense. i wouldn't worry about the update.